addons/zylann.scatter/tools/plugin.gd

```diff
@@ -277,21 +278,26 @@ static func get_scatter_child_instance(node, scatter_root):
 	return null
 
 
-func set_pattern(pattern):
-	if _pattern != pattern:
-		_pattern = pattern
-		var temp = pattern.instance()
-		# TODO This causes errors because of accessing `global_transform` outside the tree... Oo
-		# See https://github.com/godotengine/godot/issues/30445
-		var aabb = Util.get_scene_aabb(temp)
-		_pattern_margin = aabb.size.length() * 0.4
-		temp.free()
+func set_patterns(patterns):
+	if _patterns != patterns:
+		_patterns = patterns
+		var largest_aabb = AABB()
+		for pattern in patterns:
+			var temp = pattern.instance()
+			# TODO This causes errors because of accessing `global_transform` outside the tree... Oo
+			# See https://github.com/godotengine/godot/issues/30445
+			largest_aabb = largest_aabb.merge(Util.get_scene_aabb(temp))
+			temp.free()
+		_pattern_margin = largest_aabb.size.length() * 0.4
 		print("Pattern margin is ", _pattern_margin)
 
 
-func _on_Palette_pattern_selected(pattern_index):
-	var patterns = _node.get_patterns()
-	set_pattern(patterns[pattern_index])
+func create_pattern_instance():
+	return _patterns[selected_patterns[floor(randf() * selected_patterns.size())]].instance()
+
+
+func _on_Palette_patterns_selected(_pattern_index):
+	selected_patterns = _palette._item_list.get_selected_items()
```

Comment:
Think twice before you write this sequence of characters `._`
`_item_list` is private. Instead you should be able to receive selected items from the signal arguments, but you forgot to update the argument name, which should now be `selected_indices`.
This code could just become `_selected_pattern_indices = selected_indices`.

Reply:
Done.